File: sample-infrastructure/npm/src/dependency-injection/options.ts
import { IPostgresConnection } from "sample-infrastructure-persistence-orm";

export interface IOptions {
    domainCommands?: boolean;
    domainQueries?: boolean;
    postgres?: IPostgresConnection;
}

export class Options implements IOptions {
    public domainCommands?: boolean;
    public domainQueries?: boolean;
    public postgres?: IPostgresConnection;
    constructor() {
        this.domainCommands = true;
        this.domainQueries = true;
    }
}

File: sample-infrastructure/dotnet/src/dependency-injection/options.cs
using System;
using Sample.Infrastructure.Persistence.ORM;

namespace Sample.Infrastructure.DependencyInjection
{
    public interface IOptions 
    { 
        bool DomainQueries { get; set; }
        bool DomainCommands { get; set; }
        PostgresConnection Postgres { get; set; }
        Nullable<OrmType> ORM { get; set; }
    }
    
    public class Options : IOptions
    { 
        public bool DomainQueries { get; set; } = true;
        public bool DomainCommands { get; set; } = true;
        public PostgresConnection Postgres { get; set; }
        public Nullable<OrmType> ORM { get; set; }
    }
}